app.js

```diff
@@ -306,3 +311,114 @@ humburger.addEventListener('click', () => {
 cancelIcon.addEventListener('click', () => {
   menu.classList.remove('mob-menu');
 });
+
+// Implement the Validation of the form inputs
+
+const contactForm = document.querySelector('.contactForm');
+const inputName = document.querySelector('#name');
+const inputEmail = document.querySelector('#email');
+const inputComment = document.querySelector('#message');
+function putErrorContainer(input) {
+  const parent = input.parentElement;
+  const div = document.createElement('div');
+  div.className = 'error';
+  parent.appendChild(div);
+}
+
+putErrorContainer(inputName);
+putErrorContainer(inputEmail);
+putErrorContainer(inputComment);
+
+// Implement Error message on the front end
+
+function throwError(input) {
+  const errorContainer = input.parentElement;
+  const errorText = errorContainer.querySelector('div');
+  errorText.innerHTML = `Please this '${input.name}' form is required`;
+}
+
+// Implement Success message on the front end
+
+function success(input) {
+  const parent = input.parentElement;
+  const div = parent.querySelector('div');
+  div.innerHTML = '';
+}
+
+// Impement Validation function
+function inputValidation() {
+  const inputNameValue = inputName.value.trim();
+  const inputEmailValue = inputEmail.value.trim();
+  const inputCommentValue = inputComment.value.trim();
+
+  const expression = /^[a-z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-z0-9-]+(?:\.[a-z0-9-]+)*$/;
+  const IsEmailValid = expression.test(inputEmailValue);
+
+  if (inputNameValue === '') {
+    throwError(inputName);
+  } else {
+    success(inputName);
+  }
+
+  if (inputEmailValue === '') {
+    throwError(inputEmail);
+  } else if (!IsEmailValid) {
+    const div = inputEmail.parentElement.querySelector('div');
+    div.innerText = 'Please your email should be valide';
+  } else {
+    success(inputEmail);
+  }
+
+  if (inputCommentValue === '') {
+    throwError(inputComment);
+  } else {
+    success(inputComment);
+  }
+}
```

Comment:
Hello @Temmarie

May you please clarify the second comment?

Reply:
Do you mean this one?

> It'll be nice if you could tell the user why the email is invalid. So instead of your error message returning just Please your email should be valide, you could say Invalid, email should be in lowercase! .

Reply:
@Temmarie
Yes sorry, thanks ,I was asking about the first one that says `The error message should show up close to the "submit" button.`

Reply:
Oh yes, as you can see in the[ project requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/9_validate_form.md), it says;

> If the validation is not OK, you show an error message to the user near the submit button informing them of the error and the form is not sent.

Reply:
@Temmarie  thanks for your explanation, I have done everything the way you said. May you please check my changes and tell me if everything is okay?

Reply:
Hey @andersonlebon , you've done most of them, except the `Error message should show up close to the submit button`.

Reply:
@Temmarie but I have done it also. Is it in this way? :arrow_down: 

![forminput](https://user-images.githubusercontent.com/65068771/124133677-5b761100-da82-11eb-9dd9-a7b506a0eff0.png)

Reply:
Yes, right by the submit button, but change the error message to the one you used for the email